src/app/WriteClient.cpp

```diff
@@ -0,0 +1,406 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the initiator side of a CHIP Write Interaction.
+ *
+ */
+
+#include <app/InteractionModelEngine.h>
+#include <app/WriteClient.h>
+
+namespace chip {
+namespace app {
+
+CHIP_ERROR WriteClient::Init(Messaging::ExchangeManager * apExchangeMgr, InteractionModelDelegate * apDelegate,
+                             intptr_t aAppIdentifier)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    VerifyOrReturnError(apExchangeMgr != nullptr, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
do we have an 'invalid argument' for argument checking instead of 'incorrect state'?

Reply:
Done, thanks